### ISSUE TITLE
macOS focus follows keyboard not working

### ISSUE THREAD

**Issue by sshimko** [warpdotdev/Warp#3374]
### Describe the bug

When using the vision accessibility features in macOS one can have a portion of the screen appear in a magnified area of screen (this picture-in-picture that moves). With most apps, you can have this focus area move with the cursor. This does not work in Warp rendering it very difficult to use if you have vision issues and rely on this macOS feature.

### To reproduce

In macOS settings:
1. Go to Accessibility
2. Zoom
3. Set zoom style to Picture-in-Picture
4. Go to Advanced
5. In the Follow Focus section enable Follow Keyboard Focus by settings to "always" .
6. Use Warp and then try Terminal or iTerm2 or something else.

In the other terminal emulators the focus follows the cursor as you type, but not in Warp.

### Expected behavior

Focus follows the cursor the cursor as you type in Warp with the mentioned Accessibility feature enabled.

### Operating system

MacOS

### Operating system and version

13.4.1

### Shell Version

zsh 5.9

### Current Warp version

v0.2023.07.11.08.03.stable_00

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, I confirmed this only happens in Warp, not other terminals I have installed (iTerm2 and Terminal.app). This same behavior _is_ present in both Terminal.app and iTerm2 when the Secure Keyboard Entry option is enabled. If it is related, perhaps make it configurable?

### Warp Internal (ignore): linear-label:b8107fdf-ba31-488d-b103-d271c89cac3e

None

**Comment by ssam-white** [comment]
I've experienced the same issue and it also prevents me from using warp daily. I've been checking in on the progress of this issue and it hasn't been fixed. Does anyone know when this issue could be resolved?
